recipes/bacnet-stack/all/conanfile.py

```diff
@@ -0,0 +1,68 @@
+import os
+from conans import CMake, ConanFile, tools
+
+
+class BacnetStackConan(ConanFile):
+    name = "bacnet-stack"
+    license = "GPL-2.0-or-later"
+    homepage = "https://github.com/bacnet-stack/bacnet-stack/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """
+        BACnet Protocol Stack library provides a BACnet application layer,
+        network layer and media access (MAC) layer communications services."""
+    topics = ("bacnet")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+    generators = "cmake", "cmake_find_package"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
```

Comment:
See `https://github.com/conan-io/conan-center-index/pull/784#discussion_r378786929`
```suggestion
        extracted_dir = os.path.splitext(os.path.basename(self.conan_data["sources"][self.version]["url"]))[0]
```